db/db_impl_write.cc

```diff
@@ -395,9 +441,20 @@ Status DBImpl::WriteToWAL(const autovector<WriteThread::Writer*>& write_group,
   WriteBatchInternal::SetSequence(merged_batch, sequence);
 
   Slice log_entry = WriteBatchInternal::Contents(merged_batch);
+  if (concurrent) {
+    // We need to lock mutex_ since logs_ etc. might change concurrently
+    mutex_.Lock();
```

Comment:
Make sense. It would require more work since log_ (and its family) are assumed to be protected by mutex_ all over the code. But it is doable.